docs/storage-config.md

```diff
@@ -0,0 +1,267 @@
+whose backing store refers to the master PVC.
+
+
+## Populating PVCs
+
+The virtual machine startup code path should not care how the PVCs associated
+with the disks are populated. It is assumed that the PVCs have been setup ahead
+of time, either by the cloud administrator, or by the tenant user. There are a
+number of possible approaches that could be used to populate these PVCs, the
+choice of which is opaque to virt-handler, where merely needs the PVC name.
+
+This doc will outline some high level concepts, but detailed technical designs
+for them will be left for separate documents to define.  The PODs described
+below could either be things that are run prior to VM launch, to statically
+populate some PVCs with persistent data, or they could be used as part of the
+dynamic disk source described above, to create storage on-demand during VM
+startup.
+
+### HTTP upload
+
+A "HTTP upload" POD could be created that references the PVC that needs to
+be populated. This POD would run a self-contained service that just exposes a
+HTTP service that accepts a PUT request at a well known URL. Any data sent to
+this URL would be immediately written into the PVC. Thus a tenant user can
+populate an arbitrary PVC with data by simply launching a POD that references
+the this 'kubevirt-pvc-upload' image, and a PVC, then use 'curl' to PUT the
+payload.
+
+### Container repository copy
+
+A "contanier repository copy" POD could be created that references the PVC that
+needs to be populated. This POD would run a self-contained service that just
+pulls down a named image from a container repository, extracts a disk image file
+(raw / qcow2 / whatever) from this image, and copies it into the PVC. When the
+copy is complete the POD would exit
+
+
+### Container repository stream
+
+A "container repository stream" POD could be created that exposes an ISCSI server
+with one or more LUNs. The POD would run a self-contained service that just
+pulls down, one or more, named images from a container repository, extracting the
+disk image files (raw / qcow2 / whatever) from each image. The extracted images
+that then exported by the iSCSI server. The ISCSI LUNs can be directly
+referenced by a VM spec disk image, or PVCs can be created to abtract away the
+iSCSI server addresses.
+
+
+### Container image to disk adapter
+
+A "container image to disk adapter" POD could created that takes a docker
+image, and auto-creates a set of stacked qcow2 files which are populated with
+each layer in the contanier image. The POD would provide a kernel + initrd
+image that would be booted, and then run the container image main entrypoint.
+This essentially enables the "Clear Containers" usage scenario in Kubevirt by
+allowing existing docker images to be directly consumed as if they were disk
+images. Thus tenant users would never need to deal with qcow2 images directly,
+they can use normal docker tools for building images & let kubevirt boot them
+as-is.
```

Comment:
This adapter concept is a little convoluted to me. Approaching it from the user perspective will help give me clarity.

A user has a vm image, they want that image in the docker registery in a way that it can be consumed by this "container image to disk adapter". What are the steps the user has to go through to build the container image?